embulk-output-oracle/src/main/java/org/embulk/output/OracleOutputPlugin.java

```
@@ -0,0 +1,150 @@
+
+        Properties props = new Properties();
+        props.setProperty("user", oracleTask.getUser());
+        props.setProperty("password", oracleTask.getPassword());
+        props.putAll(oracleTask.getOptions());
+
+        return new OracleOutputConnector(url, props);
+    }
+
+    @Override
+    protected BatchInsert newBatchInsert(PluginTask task) throws IOException, SQLException
+    {
+        return new StandardBatchInsert(getConnector(task, true));
+    }
+    
+    @Override
+    protected ColumnSetterFactory newColumnSetterFactory(BatchInsert batch, PageReader pageReader,
+            TimestampFormatter timestampFormatter) {
+        return new OracleColumnSetterFactory(batch, pageReader, timestampFormatter);
+    }
+    
+    @Override
+    protected String generateSwapTableName(PluginTask task) throws SQLException {
+        OracleOutputConnector connector = getConnector(task, true);
+
+        int minLength = 3;
+        String tablePrefix = task.getTable();
+        if (tablePrefix.length() > MAX_TABLE_NAME_LENGTH - minLength) {
+            tablePrefix = tablePrefix.substring(0, MAX_TABLE_NAME_LENGTH - minLength);
+        }
+
+        try (OracleOutputConnection connection = connector.connect(true)) {
+            for (int i = 0; ; i++) {
+                String s = Integer.toString(i);
+                if (tablePrefix.length() + s.length() > MAX_TABLE_NAME_LENGTH) {
+                    break; 
+                }
+
+                StringBuilder sb = new StringBuilder();
+                sb.append(tablePrefix);
+                for (int j = tablePrefix.length(); j < MAX_TABLE_NAME_LENGTH - s.length(); j++) {
+                    sb.append("0");
+                }
+                sb.append(s);
+
+                String table = sb.toString();
+                if (!connection.tableExists(table)) {
+                    return table;
+                }
+                
+                if (i == Integer.MAX_VALUE) {
+                	break;
```

this looks risky. we can throw an exception.